src/ophyd_async/tango/signal/_signal.py

```diff
@@ -0,0 +1,223 @@
+    """Create a `SignalR` backed by 1 Tango Attribute/Command
+
+    Parameters
+    ----------
+    datatype:
+        Check that the Attribute/Command is of this type
+    read_trl:
+        The Attribute/Command to read and monitor
+    device_proxy:
+        If given, this DeviceProxy will be used
+    timeout:
+        The timeout for the read operation
+    name:
+        The name of the Signal
+    """
+    backend = make_backend(datatype, read_trl, read_trl, device_proxy)
+    return SignalR(backend, timeout=timeout, name=name)
+
+
+def tango_signal_w(
+    datatype: type[T],
+    write_trl: str,
+    device_proxy: DeviceProxy | None = None,
+    timeout: float = DEFAULT_TIMEOUT,
+    name: str = "",
+) -> SignalW[T]:
+    """Create a `SignalW` backed by 1 Tango Attribute/Command
+
+    Parameters
+    ----------
+    datatype:
+        Check that the Attribute/Command is of this type
+    write_trl:
+        The Attribute/Command to write to
+    device_proxy:
+        If given, this DeviceProxy will be used
+    timeout:
+        The timeout for the write operation
+    name:
+        The name of the Signal
+    """
+    backend = make_backend(datatype, write_trl, write_trl, device_proxy)
+    return SignalW(backend, timeout=timeout, name=name)
+
+
+def tango_signal_x(
+    write_trl: str,
+    device_proxy: DeviceProxy | None = None,
+    timeout: float = DEFAULT_TIMEOUT,
+    name: str = "",
+) -> SignalX:
+    """Create a `SignalX` backed by 1 Tango Attribute/Command
+
+    Parameters
+    ----------
+    write_trl:
+        The Attribute/Command to write its initial value to on execute
+    device_proxy:
+        If given, this DeviceProxy will be used
+    timeout:
+        The timeout for the command operation
+    name:
+        The name of the Signal
+    """
+    backend = make_backend(None, write_trl, write_trl, device_proxy)
+    return SignalX(backend, timeout=timeout, name=name)
+
+
+async def tango_signal_auto(
+    datatype: type[T] | None = None,
+    *,
+    trl: str,
+    device_proxy: DeviceProxy | None,
+    timeout: float = DEFAULT_TIMEOUT,
+    name: str = "",
+) -> SignalW | SignalX | SignalR | SignalRW | None:
+    try:
+        signal_character = await infer_signal_character(trl, device_proxy)
+    except RuntimeError as e:
+        if "Commands with different in and out dtypes" in str(e):
+            return None
+        else:
+            raise e
+
+    if datatype is None:
+        datatype = await infer_python_type(trl, device_proxy)
+
+    backend = make_backend(datatype, trl, trl, device_proxy)
+    if signal_character == "RW":
+        return SignalRW(backend=backend, timeout=timeout, name=name)
+    if signal_character == "R":
+        return SignalR(backend=backend, timeout=timeout, name=name)
+    if signal_character == "W":
+        return SignalW(backend=backend, timeout=timeout, name=name)
+    if signal_character == "X":
+        return SignalX(backend=backend, timeout=timeout, name=name)
```

Comment:
I'd prefer this to be private, so that people didn't end up using `tango_signal_auto` everywhere instead of `tango_signal_r(datatype)`. What do you reckon?

Reply:
We could do that although it may not have the desired effect. By default, device init will infer the existence, types, and RW character of all attributes/commands on a tango device. By not annotating any signals they can simply point a tango device to a TRL and it will automatically do tango_signal_auto for everything. I suspect this will be the case for many users that don't really care about type safety. If they don't care about type safety, they probably also don't care about encapsulation and will just use the method if it is private anyway.